src/txadmission.cpp

```diff
@@ -692,17 +765,48 @@ bool ParallelAcceptToMemoryPool(Snapshot &ss,
 
                     if (!view.HaveCoin(txin.prevout))
                     {
+                        // Are inputs missing because we already have the ptx?
+                        for (size_t out = 0; out < tx->vout.size(); out++)
+                        {
+                            // Optimistically just do efficient check of cache for
+                            // outputs.
+                            if (pcoinsTip->HaveCoinInCache(COutPoint(tx->GetHash(), out)))
+                            {
+                                // return state.Invalid(false, REJECT_DUPLICATE,
+                                //                     "txn-already-known");
+                                // debugger->AddInvalidReason("txn-already-known");
+                                txnAlreadyKnown = true;
+                                break;
+                            }
+                        }
```

Comment:
seems this loop is no longer needed since you took the debugger section out "if (txnAlreadyKnown)"  ?